src/browser/provider/index.ts

```diff
@@ -240,6 +244,17 @@ export default class BrowserProvider {
         await browserTools.maximize(this._getWindowDescriptor(browserId));
     }
 
+    private async _ensureRetryTestPagesWarning (browserId: string): Promise<void> {
```

Comment:
I guess it should be a little bit more complicated:
* if connection is not secure, we should raise an error and abort test execution. Secure connection is a option usage requirement. 
Customer can fix this error by turning on the secure connection (specify 'localhost' as a 'host' parameter value or run TestCafe via HTTPS).

* if one of used browsers is not supported ServiceWorker, we should raise a warning. Customer cannot fix it. Also, it allows to run tests in such way: `testcafe chrome, ie tests --retry-test-pages`.

Reply:
Agree with @miherlosev in general, but Safari will adds complexity here: it supports Service Workers only from version 12 and newer